tests/functional/test_download.py

```diff
@@ -666,7 +666,9 @@ def test_download_exit_status_code_when_blank_requirements_file(script):
     Test download exit status code when blank requirements file specified
     """
     script.scratch_path.joinpath("blank.txt").write_text("\n")
-    script.pip('download', '-r', 'blank.txt')
+    result = script.pip('download', '-r', 'blank.txt', expect_error=True)
+    print(result)
+    assert result.returncode == ERROR
```

Comment:
@deveshks if that test did not return an error before it should continue to do so.

Reply:
On the one hand, I think that'd be proper behavior. OTOH, I do agree that it's a change from the existing behavior and should go through regular scrutiny. :)

Reply:
I suppose if someone took the trouble to write a test, there must be a use case somewhere :) In any case if that needs changing it should be in another PR.

Reply:
So what is the final verdict here, should I leave this change as-is, or I should revert just the `assert result.returncode == ERROR` (`script.pip('download', '-r', 'blank.txt', expect_error=True)` will have to be here since we return `ERROR` status.

Reply:
My proposal is to keep the test as is and `return SUCCESS` in run() even if the downloaded list is empty. An error would be raised as an exception anyway.

Reply:
That's a good idea. Have applied it in latest commit https://github.com/pypa/pip/pull/8265/commits/e8b842389c94111f510d372ca834a6d4a123ae0b

Reply:
> I suppose if someone took the trouble to write a test, there must be a use case somewhere :) In any case if that needs changing it should be in another PR.

Now that all the `run` funtions in `pip._internal.commands.*` are returning a error code, is it a good idea to create a blanket PR which revisits the tests, and add `returncode` asserts to utilize them, which includes the change which was made here?

Reply:
@deveshks I don't think the tests need changing? The commands already have a proper exit code. What you can do though is double check that all run() have the proper typing and then update [base_command](https://github.com/pypa/pip/blob/7d875faff908e2131108e1bcc71475ecc031098e/src/pip/_internal/cli/base_command.py#L195-L199).

Reply:
Yes, I double checked and atleast all `run` commands have return type `int`, so I think the base_command can be updated.

Also I didn't mean to change the tests, I meant that since we are using exit codes in all commands, we can update the tests for them to explicitly check `result.returncode == *` in them, similar to how https://github.com/pypa/pip/blob/master/tests/functional/test_check.py have returncode checks in all their tests.

Updating the basecommand and checking the return code for atleast a few tests can happen in one PR, updating the return codes for other tests in subsequent PRs.

Reply:
>  What you can do though is double check that all run() have the proper typing and then update [base_command](https://github.com/pypa/pip/blob/7d875faff908e2131108e1bcc71475ecc031098e/src/pip/_internal/cli/base_command.py#L195-L199).

@sbidoul , in the above statement, were you talking about something like this? (https://github.com/pypa/pip/compare/master...deveshks:remove-isinstance-fixme)  If yes, I can go ahead and create a PR for the same.

Reply:
PR made at https://github.com/pypa/pip/pull/8336